ocs_ci/ocs/fiojob.py

```diff
@@ -548,6 +558,10 @@ def is_storage_reclaimed():
         # Without checksum, we just need to make sure that data were deleted
         # and wait for this to happen to avoid conflicts with tests executed
         # right after this one.
-        delete_fio_data(fio_job_file, is_storage_reclaimed)
+        if delete_fio_data_after_test:
+            delete_fio_data(fio_job_file, is_storage_reclaimed)
+        # Wait until the end of the tests, and then delete the data.
+        else:
+            request.addfinalizer(lambda: delete_fio_data(fio_job_file, is_storage_reclaimed))
```

Comment:
It might be good to add log info that `request` object have to be provided as parameter. It might help to debug for people that use this function without providing `request` object which has default value set to `None`.